### PR TITLE
[TECH]: ship comet's communication service dummy with Heroic

### PR DESCRIPTION
**Pull request by imLinguin**
This will give us more control over the service we install in prefix.

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

File: meta/downloadHelperBinaries.ts [hunk 1]
```diff
@@ -108,7 +108,7 @@ async function downloadLegendary() {
 }
 
 async function downloadGogdl() {
-  return downloadGithubAssets(
+  await downloadGithubAssets(
```
Comment (arielj) [inline meta/downloadHelperBinaries.ts:111]: I understand these were `return` on purpose cause it's awaiting all promises later

https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/829c8dc68d3251f4eaa75de1d0f70f80678ccbaf/meta/downloadHelperBinaries.ts#L220

Comment (imLinguin) [reply meta/downloadHelperBinaries.ts:111]: Those functions return a Promise themselves this doesn't change anything really
Here, I'm printing a date before and after awaiting promises

```
Downloading: [ 'legendary', 'gogdl', 'nile', 'comet' ]
Downloading legendary for linux x64 from https://github.com/Heroic-Games-Launcher/legendary/releases/download/0.20.35/legendary_linux_x86_64
Downloading legendary for darwin x64 from https://github.com/Heroic-Games-Launcher/legendary/releases/download/0.20.35/legendary_macOS_x86_64
Downloading legendary for win32 x64 from https://github.com/Heroic-Games-Launcher/legendary/releases/download/0.20.35/legendary_windows_x86_64.exe
Downloading legendary for darwin arm64 from https://github.com/Heroic-Games-Launcher/legendary/releases/download/0.20.35/legendary_macOS_arm64
Downloading gogdl for linux x64 from https://github.com/Heroic-Games-Launcher/heroic-gogdl/releases/download/v1.1.1/gogdl_linux_x86_64
Downloading gogdl for darwin x64 from https://github.com/Heroic-Games-Launcher/heroic-gogdl/releases/download/v1.1.1/gogdl_macOS_x86_64
Downloading gogdl for win32 x64 from https://github.com/Heroic-Games-Launcher/heroic-gogdl/releases/download/v1.1.1/gogdl_windows_x86_64.exe
Downloading gogdl for darwin arm64 from https://github.com/Heroic-Games-Launcher/heroic-gogdl/releases/download/v1.1.1/gogdl_macOS_arm64
Downloading nile for linux x64 from https://github.com/imLinguin/nile/releases/download/v1.1.1/nile_linux_x86_64
Downloading nile for darwin x64 from https://github.com/imLinguin/nile/releases/download/v1.1.1/nile_macOS_x86_64
Downloading nile for win32 x64 from https://github.com/imLinguin/nile/releases/download/v1.1.1/nile_windows_x86_64.exe
Downloading nile for darwin arm64 from https://github.com/imLinguin/nile/releases/download/v1.1.1/nile_macOS_arm64
Downloading GalaxyCommunication for win32 x64 from https://github.com/imLinguin/comet/releases/download/v0.1.2/GalaxyCommunication-dummy.exe
2024-08-10T14:54:34.189Z
Done downloading nile for darwin arm64
Done downloading legendary for linux x64
Done downloading GalaxyCommunication for win32 x64
Downloading comet for linux x64 from https://github.com/imLinguin/comet/releases/download/v0.1.2/comet-x86_64-unknown-linux-gnu
Downloading comet for darwin x64 from https://github.com/imLinguin/comet/releases/download/v0.1.2/comet-x86_64-apple-darwin
Downloading comet for win32 x64 from https://github.com/imLinguin/comet/releases/download/v0.1.2/comet-x86_64-pc-windows-msvc.exe
Downloading comet for darwin arm64 from https://github.com/imLinguin/comet/releases/download/v0.1.2/comet-aarch64-apple-darwin
Done downloading gogdl for darwin x64
Done downloading comet for darwin x64
Done downloading comet for win32 x64
Done downloading gogdl for darwin arm64
Done downloading legendary for darwin x64
Done downloading legendary for darwin arm64
Done downloading gogdl for linux x64
Done downloading gogdl for win32 x64
Done downloading nile for darwin x64
Done downloading nile for linux x64
Done downloading nile for win32 x64
Done downloading legendary for win32 x64
Done downloading comet for linux x64
Done downloading comet for darwin arm64
2024-08-10T14:54:39.277Z

```

File: src/backend/launcher.ts [hunk 2]
```diff
@@ -372,26 +374,38 @@ async function prepareWineLaunch(
...
+        'bin/x64/win32/GalaxyCommunication.exe'
       )
+
+      const winePrefix = gameSettings.winePrefix
+      let driveLocation =
+        'drive_c/ProgramData/GOG.com/Galaxy/redists/GalaxyCommunication.exe'
+      if (gameSettings.wineVersion.type === 'proton') {
+        driveLocation = join('pfx', driveLocation)
+      }
+      const communicationDest = join(winePrefix, driveLocation)
```
Comment (arielj) [inline src/backend/launcher.ts:390]: Maybe this can be replaced with the `getWinePath` method? https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/17fd9fbf35c37d55daafa3fc3021db19f3c93dd7/src/backend/launcher.ts#L1326

not a blocker though

**Review by arielj — APPROVED** [review]
looks good to me, I added a suggestion but not a blocker

**Comment by imLinguin** [comment]
Added a optional bypass for `PROTON_LOG` useful when we want to ensure we'll read the output. I've noticed this by accident just now, seems that it could cause issues with GOG ISI setup as well

File: meta/downloadHelperBinaries.ts [hunk 3]
```diff
@@ -139,7 +139,18 @@ async function downloadNile() {
...
+    'imLinguin/comet',
+    RELEASE_TAGS['comet'],
+    {
+      x64: {
+        win32: 'GalaxyCommunication-dummy.exe'
+      },
+      arm64: {}
+    }
+  )
+  await downloadGithubAssets(
```
Comment (CommandMC) [inline meta/downloadHelperBinaries.ts:153]: The two promises being awaited here should ideally run in parallel to fully utilize the network bandwidth

```ts
return Promise.all([
  downloadGithubAssets('GalaxyCommunication', /* ... */ ),
  downloadGithubAssets('comet', /* ... */ )
])

File: meta/downloadHelperBinaries.ts [hunk 4]
```diff
@@ -126,7 +126,7 @@ async function downloadGogdl() {
 }
 
 async function downloadNile() {
-  return downloadGithubAssets('nile', 'imLinguin/nile', RELEASE_TAGS['nile'], {
+  await downloadGithubAssets('nile', 'imLinguin/nile', RELEASE_TAGS['nile'], {
```
Comment (CommandMC) [inline meta/downloadHelperBinaries.ts:129]: Why was this changed?

**Review by flavioislima — APPROVED** [review]
